$NetBSD: patch-aj,v 1.5 2011/09/04 22:05:39 dholland Exp $

- Rearrange position of some static functions to allow them to
self-define and compile under GCC-4.

- fix void main.

--- main.c.orig	1993-01-12 18:59:36.000000000 +0000
+++ main.c
@@ -23,13 +23,36 @@
 
 Widget		top_form;
 
+extern AppData app_data;
+
+static void
+chk_debug(argc, argv)
+	int	argc;
+	char	**argv;
+{
+	int	i;
+
+	/* ugly hack */
+	for (i = 1; i < argc; i++) {
+		if (strcmp(argv[i], "-debug") == 0) {
+			app_data.debug = True;
+			break;
+		}
+	}
+}
+
 void
+usage() {
+	(void) fprintf(stderr, "usage: xcdplayer [-debug]\n");
+
+	exit(1);
+}
+
+int
 main(argc, argv)
 	int		argc;
 	char		**argv;
 {
-	static void	chk_debug();
-
 	chk_debug(argc, argv);
 
 	/* gets resources, & creates main form */
@@ -55,26 +78,3 @@ main(argc, argv)
 
 	exit(0);
 }
-
-static void
-chk_debug(argc, argv)
-	int	argc;
-	char	**argv;
-{
-	int	i;
-
-	/* ugly hack */
-	for (i = 1; i < argc; i++) {
-		if (strcmp(argv[i], "-debug") == 0) {
-			debug = True;
-			break;
-		}
-	}
-}
-
-void
-usage() {
-	(void) fprintf(stderr, "usage: xcdplayer [-debug]\n");
-
-	exit(1);
-}
